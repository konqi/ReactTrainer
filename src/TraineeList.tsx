import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow
} from "@material-ui/core";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import * as React from "react";
import { useContext } from "react";
import { ApplicationContext } from "./App";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: "100%",
      marginTop: theme.spacing(3),
      overflowX: "auto"
    },
    table: {
      minWidth: "85%"
    }
  })
);

export const TraineeList = () => {
  const classes = useStyles();

  const { state } = useContext(ApplicationContext);

  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell align="right">Preis</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {state!.trainees.map(trainee => (
            <TableRow key={trainee.name}>
              <TableCell component="th" scope="row">
                {trainee.name}
              </TableCell>
              <TableCell align="right">{trainee.price}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  );
};
