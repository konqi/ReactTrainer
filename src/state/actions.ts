export enum ActionType {
  ADD_TRAINEE
}
