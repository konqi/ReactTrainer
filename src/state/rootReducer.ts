import {combineReducers} from 'redux'
import {reduceTrainees} from './trainees'
import {reduceUi} from './ui'

export const rootReducer = combineReducers({
  trainees: reduceTrainees,
  ui: reduceUi,
})
