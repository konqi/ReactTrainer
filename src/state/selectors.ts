import {ApplicationState} from '.'

export const selectTraineeById = (state: ApplicationState, traineeId: string) =>
  state.trainees[traineeId]
