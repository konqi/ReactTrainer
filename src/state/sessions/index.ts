export {
  createAddSessionAction as createAddTrainingAction,
} from './sessionActions'
export {initialSessionState} from './sessionReducer'
