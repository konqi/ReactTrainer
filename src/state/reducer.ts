import { ActionType } from "./actions";

export interface Action {
  type: ActionType;
  payload: Trainee;
}

export interface Trainee {
  name: string;
  price: number;
}

export interface ApplicationState {
  trainees: Trainee[];
}

export const initialApplicationState: ApplicationState = {
  trainees: []
};

export const reducer = (
  state: ApplicationState,
  action: Action
): ApplicationState => {
  switch (action.type) {
    case ActionType.ADD_TRAINEE:
      const newTrainees = [...state.trainees, action.payload];
      return { ...state, trainees: newTrainees };
    default:
      throw new Error();
  }
};
