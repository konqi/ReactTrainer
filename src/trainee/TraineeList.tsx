import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
} from '@material-ui/core'
import {createStyles, makeStyles, Theme} from '@material-ui/core/styles'
import * as React from 'react'
import {useDispatch, useSelector} from 'react-redux'
import {ApplicationState} from '../state'
import {createUiNavigateAction} from '../state/ui/uiActions'
import {Page} from '../types/page'
import {Trainee} from '../types/Trainee'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '100%',
      marginTop: theme.spacing(3),
      overflowX: 'auto',
    },
    table: {
      minWidth: '85%',
    },
  })
)

export const TraineeList: React.FC = () => {
  const classes = useStyles()
  const trainees = useSelector<ApplicationState, Trainee[]>(
    state => state.trainees
  )
  const dispatch = useDispatch()
  const openTrainee = (traineeId: string) => {
    dispatch(createUiNavigateAction(Page.Trainee, {traineeId}))
  }

  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell align="right">Preis</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {trainees.map(trainee => (
            <TableRow
              key={trainee.id}
              hover
              onClick={() => openTrainee(trainee.id)}
            >
              <TableCell component="th" scope="row">
                {trainee.name}
              </TableCell>
              <TableCell align="right">{trainee.price}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  )
}
