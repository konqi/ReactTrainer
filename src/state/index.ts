import {TraineeState, TraineeFSAs} from './trainees/types'
import {initialTraineeState} from './trainees'
import {UiState, UiFSAs} from './ui/types'
import {TrainingFSAs} from './trainings/types'
import {initialUiState} from './ui/uiReducer'
import {createStore, applyMiddleware} from 'redux'
import {rootReducer} from './rootReducer'
import createSagaMiddleware from 'redux-saga'
import {rootSaga} from './sagas/rootSaga'

export interface ApplicationState {
  trainees: TraineeState
  ui: UiState
}

export const initialApplicationState: ApplicationState = {
  trainees: initialTraineeState,
  ui: initialUiState,
}

export type Action = TraineeFSAs | TrainingFSAs | UiFSAs

const sagaMiddleware = createSagaMiddleware()

export const store = createStore(
  rootReducer,
  initialApplicationState,
  applyMiddleware(sagaMiddleware)
)

sagaMiddleware.run(rootSaga)
