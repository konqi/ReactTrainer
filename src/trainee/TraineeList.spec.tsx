import * as React from 'react'
import {TraineeListConnected} from './TraineeList'
import {render} from '@testing-library/react'
import {ApplicationState} from '../state'
import {createStore} from 'redux'
import {rootReducer} from '../state/rootReducer'
import {Provider} from 'react-redux'

describe('snapshot tests', () => {
  test.each`
    description      | trainees
    ${'no trainee'}  | ${[]}
    ${'one trainee'} | ${[{id: 0, name: 'Tester', price: 1500}]}
  `('should show TraineeList with $description', ({description, trainees}) => {
    const state: Pick<ApplicationState, 'trainees'> = {
      trainees,
    }
    const store = createStore(rootReducer, state)

    const {baseElement, unmount} = render(
      <Provider store={store}>
        <TraineeListConnected />
      </Provider>
    )

    expect(baseElement).toMatchSnapshot()

    unmount()
  })
})
