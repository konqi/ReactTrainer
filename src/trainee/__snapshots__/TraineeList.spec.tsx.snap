// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests should show TraineeList with no trainee 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
    >
      <table
        class="MuiTable-root makeStyles-table-2"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              Preis
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        />
      </table>
    </div>
  </div>
</body>
`;

exports[`snapshot tests should show TraineeList with one trainee 1`] = `
<body>
  <div />
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-49 MuiPaper-rounded"
    >
      <table
        class="MuiTable-root makeStyles-table-50"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              Preis
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-hover"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              Tester
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1500
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;
